platform-api/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/DtoConverter.java

```diff
@@ -158,170 +126,109 @@ public static ItemReference toItemReference(FolderEntry folder,
-        }
-
-        try {
-            dto.withVisibility(project.getVisibility());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
-        }
+    /**
+     * The method tries to provide as much as possible information about project.If get error then save information about error
+     * with 'problems' field in ProjectConfigDto.
+     *
+     * @param project
+     *         project from which we need get information
+     * @param serviceUriBuilder
+     *         service for building URI
+     * @return an instance of {@link ProjectConfigDto}
+     * @throws InvalidValueException
+     */
+    public static ProjectConfigDto toProjectConfig(Project project,
+                                                   UriBuilder serviceUriBuilder) throws ForbiddenException,
+                                                                                        ServerException,
+                                                                                        ProjectTypeConstraintException,
+                                                                                        ValueStorageException {
+        String name = project.getName();
+        String path = project.getPath();
+
+        ProjectConfigDto projectConfigDto = DtoFactory.getInstance().createDto(ProjectConfigDto.class);
+
+        projectConfigDto.withName(name).withPath(path);
 
         try {
-            dto.withContentRoot(project.getContentRoot());
-        } catch (ServerException e) {
-            dto.getProblems().add(createProjectProblem(dtoFactory, e));
+            ProjectConfig config = project.getConfig();
+
+            projectConfigDto.withDescription(config.getDescription());
+
+            projectConfigDto.withMixins(config.getMixins());
+
+            projectConfigDto.withAttributes(config.getAttributes());
+
+            List<ProjectConfigDto> modules = config.getModules()
+                                                   .stream()
+                                                   .map(DtoConverter::toProjectConfigDto)
+                                                   .collect(Collectors.toList());
+            projectConfigDto.withModules(modules);
+            projectConfigDto.withContentRoot(config.getContentRoot());
+
+            projectConfigDto.withType(config.getType());
+            projectConfigDto.withSource(toSourceDto(config.getSource()));
+            projectConfigDto.setContentRoot(project.getContentRoot());
+        } catch (ServerException | ValueStorageException | ProjectTypeConstraintException exception) {
+            projectConfigDto.withType(BaseProjectType.ID).withType("blank");
+            ProjectProblem projectProblem = DtoFactory.getInstance()
+                                                      .createDto(ProjectProblem.class)
+                                                      .withCode(1)
```

Comment:
In this case i can't explain you why we use '1'. I don't write this code, i just ported it and decided don't change it, and i couldn't find any java doc which would explain why we use '1', as project problem code.